src/network/networkHandler.ts

```diff
@@ -477,16 +438,17 @@ async function handleOnDataMessage(d: OnDataArgs, overworld: Overworld): Promise
   currentlyProcessingOnDataMessage = d;
   const { payload, fromClient } = d;
   const type: MESSAGE_TYPES = payload.type;
+
   const { underworld } = overworld;
   if (!underworld) {
     console.error('Cannot handleOnDataMessage, underworld does not exist');
     return;
   }
   logHandleOnDataMessage(type, payload, fromClient, underworld);
-  // Get player of the client that sent the message 
-  const fromPlayer = globalThis.numberOfHotseatPlayers > 1 ? underworld.players[underworld.hotseatCurrentPlayerIndex] : underworld.players.find(p => p.clientId == fromClient);
+
+  const fromPlayer = getFromPlayerViaClientId(fromClient, underworld);
```

Comment:
This doesn't account for hotseat does it?

Reply:
This function should only be used in network messages to find out who the message came from, and we had a discord conversation about how this affects hotseat:

> SoulMuncher — 01/17/2024 4:24 PM
Right I was just wondering if theres ever a situation where the fromPlayer could ever be a player not currently taking their turn (maybe some delay in send/recieve messages or some game event or something), in which case this would return the current hotseat player as the "fromPlayer" when that isn't true.
But I don't think there IS a situation where that happens. So this should work fine for finding the fromPlayer. And then if we want to reference another player, that usually gets sent through the payload, so that's taken care of too.

> JordanO — 01/17/2024 4:32 PM
right, in singleplayer the wspie message gets invoked immediately since there isn't a network to send a message over

It should be fine, since network messages can never come from an inactive player in hotseat. It does feel a little weird, though.